src/MPDevice.cpp

```diff
@@ -1388,17 +1388,17 @@ quint16 MPDevice::getNumberOfPages(void)
 {
     if(get_flashMbSize() >= 16)
     {
-        return 256*get_flashMbSize();
+        return static_cast<quint16>(256*get_flashMbSize());
     }
     else
     {
-        return 512*get_flashMbSize();
+        return static_cast<quint16>(512*get_flashMbSize());
     }
 }
 
 quint16 MPDevice::getFlashPageFromAddress(const QByteArray &address)
 {
-    return (((quint16)address[1] << 5) & 0x1FE0) | (((quint16)address[0] >> 3) & 0x001F);
+    return (static_cast<quint16>(address[1] << 5) & 0x1FE0) | (static_cast<quint16>(address[0] >> 3) & 0x001F);
```

Comment:
) parenthesis wronly placed:
`return (static_cast<quint16>(address[1]) << 5` instead of `return (static_cast<quint16>(address[1] << 5)`

Same for `(static_cast<quint16>(address[0] >> 3)`